src/appleseed/renderer/kernel/shading/shadingpoint.h

```diff
@@ -611,9 +611,9 @@ inline const foundation::Vector3d& ShadingPoint::get_offset_point(const foundati
     assert(m_members & HasRefinedPoints);
 
     return
-        foundation::dot(m_asm_geo_normal, direction) > 0.0
-            ? m_front_point
-            : m_back_point;
+        foundation::dot(m_asm_inst_geo_normal, direction) > 0.0
```

Comment:
What space is `direction` in? It should be in assembly instance space (and it should be documented as such in the function _declaration_). However, if it's in world space, then we have a bug here.

Reply:
It's in assembly instance space

Reply:
The only call is the following (with direction being defined on the first line).
` 
// Transform the ray direction to assembly instance space.
        output_ray.m_dir = assembly_instance_transform.vector_to_local(input_ray.m_dir);

        // Compute the ray origin in assembly instance space.
        if (parent_sp &&
            parent_sp->get_assembly_instance().get_uid() == assembly_instance.get_uid() &&
            parent_sp->get_object_instance().get_ray_bias_method() == ObjectInstance::RayBiasMethodNone)
        {
            // The caller provided the previous intersection, and we are about
            // to intersect the assembly instance that contains the previous
            // intersection. Use the properly offset intersection point as the
            // origin of the child ray.
            output_ray.m_org = parent_sp->get_offset_point(output_ray.m_dir);
`